topi/python/topi/cuda/conv2d_nchw.py

```diff
@@ -1,544 +0,0 @@
-            s[temp_G].compute_inline()
-            i, ic, h, w = s[temp_G].op.axis
-            s[temp_G].split(w, factor=4)
-            temp_R = s.cache_write(temp_G, "global")
-            temp_S = s.cache_read(temp_R, "shared", [temp_G])
-        else:
-            s[temp].compute_inline()
-            temp_S = s.cache_read(temp, "shared", [Output])
-            temp_R = temp_S
-
-        Filter_S = s.cache_read(Filter, "shared", [Output])
-
-        if Output.op in s.outputs:
-            Out = Output
-            Out_L = s.cache_write(Out, "local")
-        else:
-            Out = outs[0].op.output(0)
-            s[Output].set_scope("local")
-            Out_L = Output
-
-        if util.get_const_int(Filter.shape[3]) == 7 or (util.get_const_int(Output.shape[2] == 224) and flag < 128):
-            conv2d_224_3_64(s, temp, temp_R, temp_S, Filter_S, Out, Out_L, flag)
-        elif 128 < flag < 512:
-            conv2d_56_64_128(s, temp, temp_R, temp_S, Filter_S, Out, Out_L, flag)
-        elif flag >= 512:
-            conv2d_14_256_256(s, temp, temp_R, temp_S, Filter, Filter_S, Out, Out_L)
-        else:
-            conv2d_56_64_64(s, Filter, temp_S, Filter_S, Out, Out_L)
-
-    scheduled_ops = []
-
-    def traverse(OP):
-        """Traverse operators from computation graph"""
-        # inline all one-to-one-mapping operators except the last stage (output)
-        if tag.is_broadcast(OP.tag):
-            if OP not in s.outputs:
-                s[OP].compute_inline()
-            for tensor in OP.input_tensors:
-                if tensor.op.input_tensors and tensor.op not in scheduled_ops:
-                    traverse(tensor.op)
-        # schedule conv2d
-        if 'conv2d_nchw' in OP.tag:
-            temp = OP.input_tensors[0]
-            Filter = OP.input_tensors[1]
-            if isinstance(Filter.op, tvm.tensor.ComputeOp) and 'dilate' in Filter.op.tag:
-                s[Filter].compute_inline()
-            Output = OP.output(0)
-            schedule(temp, Filter, Output)
-
-        scheduled_ops.append(OP)
-
-    traverse(outs[0].op)
-    return s
-
-
-@generic.schedule_conv2d_nchw.register(["cuda", "gpu"])
-def schedule_conv2d_nchw(outs):
-    """Schedule for conv2d_nchw.
-
-    Parameters
-    ----------
-    outs: Array of Tensor
-        The computation graph description of conv2d_nchw
-        in the format of an array of tensors.
-
-    Returns
-    -------
-    s: Schedule
-        The computation schedule for conv2d_nchw.
-    """
-    target = tvm.target.current_target()
-    if target.target_name == "cuda" and "cudnn" in target.libs:
-        return topi.generic.schedule_extern(outs)
-
-    outs = [outs] if isinstance(outs, tvm.tensor.Tensor) else outs
-    batch_size = util.get_const_int(outs[0].op.output(0).shape[0])
-    if batch_size > 1:
-        raise RuntimeError("Batch size: %d is too large for this schedule" % batch_size)
-    return  schedule_conv2d_small_batch(outs)
```

Comment:
I'm not familiar with how autotvm works, but is it a good idea to remove everything here? Can we use the existing schedule as a fallback when we don't want to train from scratch?

Reply:
The old schedule is incorrect for some shapes (You can see many error reports in the forum).
Its style is also not recommanded after we have autotvm. Their performance is also poor (You can see many reports in NNVM issues before the merge). So we should remove it.

In this pr, we can cover the shapes in (resnet/vgg/inception/mobilenet). The parameters tuned on 1080ti is faster than old schedule on most cuda devices.

For other shapes (including batch), current autotvm template supports naive fallback compilation, but it will be very slow. Warning messages will show if tvm meet un-tuned shapes.

Later I can add smarter fallback strategy, but it will be based on autotvm.

Reply:
ok. I'm aware of this schedule's limitation. Thanks for the detailed comment.